File: code-test/solutions/python/day_001.py
def solution(numbers, target):
    # TODO: O(n)으로 풀어보세요.
    # 힌트: dict에 이미 본 숫자와 인덱스를 저장합니다.

    dict = {}

    for z in range(len(numbers)) :
        if(target - numbers[z] in dict) :
            i = dict[target-numbers[z]]
            j = z
            
            break;
            
            
        dict[numbers[z]] = z


    return [i, j]

File: code-test/solutions/java/Day001.java
import java.util.*;

public class Day001 {
    public static int[] solution(int[] numbers, int target) {
        // TODO: O(n)으로 풀어보세요.
        // 힌트: HashMap에 이미 본 숫자와 인덱스를 저장합니다.

        HashMap<Integer, Integer> hm = new HashMap<>();
        int i = 0;
        int j = 0;
        for (int z=0; z<numbers.length; z++) {
            if(hm.containsKey(target - numbers[z])) {
                i = hm.get(target - numbers[z]);
                j = z;

                break;
            } else {
                hm.put(numbers[z], z);
            }

        }
        
        return new int[] {i, j};
    }
}
